lib/dir.go

```diff
@@ -80,6 +83,58 @@ func CopyThing(srcpath, destpath string) error {
 	if srcInfo.IsDir() {
 		return DirCopy(destpath, srcpath)
 	} else {
-		return FileCopy(destpath, srcpath)
+		return FileCopy(destpath, srcpath, nil, nil, nil)
 	}
 }
+
+// Chmod changes file permissions
+func Chmod(perms, destpath string) error {
```

Comment:
1. Why is setting perms on directory not supported?
2. You do however need to check for destinfo.Mode()&ModeSymlink and refuse to chmod in that case.